edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/EdgeAgentDockerModule.cs

```diff
@@ -21,9 +23,24 @@ public override bool Equals(IModule<DockerConfig> other)
                 return false;
             if (ReferenceEquals(this, other))
                 return true;
-            return string.Equals(this.Name, other.Name) &&
-                string.Equals(this.Type, other.Type) &&
-                string.Equals(this.Config.Image, other.Config.Image);
+
+            if (!string.Equals(this.Name, other.Name) ||
+                !string.Equals(this.Type, other.Type) ||
+                !string.Equals(this.Config.Image, other.Config.Image))
+                return false;
+
+            IDictionary<string, string> labels = other.Config.CreateOptions?.Labels ?? new Dictionary<string, string>();
+            if (!labels.TryGetValue(Core.Constants.Labels.CreateOptions, out string createOptions) ||
```

Comment:
In what situation, it can't get create options from label?

Reply:
It will be missing when 1st time start up by iotedged, right?

Reply:
Correct, the first time agent starts it won't have the labels, because the "bootstrap" container is created independently by iotedged. And you raise a good point...

Normally the config.yaml will point to the vanilla 1.0 agent or some other stable agent that won't change. The agent specified in the deployment will be different, so the old agent (if it has my changes) will tell iotedged to create the new container with the right labels. The unlabeled container will only run for a very short time and everything will be fine.

But if the config.yaml and the deployment specify the same agent image, then the agent won't be updated when it reads the deployment config, and the unlabeled container will continue to run. This is a problem. To fix it, I think I'll need to make an additional change to iotedged (separate PR) so that it labels the bootstrap container too. Both label values can probably just be empty JSON objects ("{}").